rosflight_sim/src/sil_board.cpp

```diff
@@ -490,23 +505,155 @@ void SIL_Board::RCCallback(const rosflight_msgs::RCRaw& msg)
 }
 
 
-bool SIL_Board::gnss_present() { return false; }
+bool SIL_Board::gnss_present() { return true; }
 void SIL_Board::gnss_update() {}
 
+static void lla2ecef(double lat, double lon, double alt, double ecef[3]) {
+  static constexpr double A = 6378137.0; // WGS-84 Earth semimajor axis (m)
+  static constexpr double B = 6356752.314245; // Derived Earth semiminor axis (m)
+  static constexpr double F = (A - B) / A; // Ellipsoid Flatness
+
+  double sinp = std::sin(lat);
+  double cosp = std::cos(lat);
+  double sinl = std::sin(lon);
+  double cosl = std::cos(lon);
+  double e2 = F * (2.0 - F);
+  double v = A / std::sqrt(1.0 - e2 * sinp * sinp);
+
+  ecef[0] = (v + alt) * cosp * cosl;
+  ecef[1] = (v + alt) * cosp * sinl;
+  ecef[2] = (v * (1.0 - e2) + alt) * sinp;
+}
+
 rosflight_firmware::GNSSData SIL_Board::gnss_read()
 {
-    rosflight_firmware::GNSSData out;
-    return out;
+  // We are going to pretend the earth is perfectly round
+  static constexpr double R = 6378145.0f;
+
+  rosflight_firmware::GNSSData out;
+  out.fix_type = rosflight_firmware::GNSSFixType::GNSS_FIX_TYPE_FIX;
+  out.time_of_week = GZ_COMPAT_GET_SIM_TIME(world_).Double()*1000;
+  out.time = GZ_COMPAT_GET_SIM_TIME(world_).Double();
+  out.nanos = (GZ_COMPAT_GET_SIM_TIME(world_).Double() - out.time)*1e9;
+
+  GazeboPose current_state_NWU = GZ_COMPAT_GET_WORLD_POSE(link_);
```

Comment:
What is NWU? That coordinate frame doesn't exist... :-)

https://www.ros.org/reps/rep-0103.html#id19

Reply:
Okay, okay I'll switch to ENU.  But not this commit.  :)  

Let's wait for this to land, and then let's have another (or maybe just continue our discussion about it)